public_website/src/ui/components/header/Header.tsx

```diff
@@ -44,17 +47,45 @@ type HeaderNavigationItemProps = {
   }
 }
 
+const findCollectionIdByPath = (
+  path: string,
+  collections: HeaderNavigationItemProps[]
+): number | null => {
+  for (const collection of collections) {
+    const { megaMenu } = collection
+    if (megaMenu.primaryListItems) {
```

Comment:
Pour éviter de répéter l'algo de `find` on peut imaginer que cette fonction 
const findXXXInMenu = (collection) => (item) =>
if (items) {
      for (const item of items) {
        if (item.URL.trim() === path) {
          return collection.id
        }
      }
    }

 soit passée en params à `findCollectionIdByPath` et injectée L:87.